### PR TITLE
EOS-24086: Purge fails on reset due to less timeout

### PR DESCRIPTION
**Pull request by selvakumaar5496**
# Problem Statement
- https://jts.seagate.com/browse/EOS-24086

# Design
-  Call purge until the rc is 0.

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: N
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]: N
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [x] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page
Not required, no change in signature.

### CODE REVIEW

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 1]
```diff
@@ -402,7 +402,7 @@ def delete(self, admin_id: str, message_type: str):
                 Log.error(f"MessageBusError: {errors.ERR_OP_FAILED} Unable" \
                     f" to delete messages for message type {message_type}" \
                     f" using admin {admin} after {retry_count} retries")
-                raise MessageBusError(errors.ERR_OP_FAILED,\
+                return MessageBusError(errors.ERR_OP_FAILED,\
```
Comment (sachinpunadikar) [inline py-utils/src/utils/message_bus/message_broker_collection.py:405]: Will this change the signature of this method ? Will it impact S3 ?

Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:405]: No, it wont.

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/src/setup/utils.py [hunk 2]
```diff
@@ -294,7 +294,10 @@ def reset():
                 for message_type in message_types_list:
                     producer = MessageProducer(producer_id=message_type, \
                         message_type=message_type, method='sync')
-                    producer.delete()
+                    while True:
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:297]: This cant be an indefinite cycle. Code may hang. you should try for retry_count times and fail if it did not succeed.

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 1 above)
Comment (ujjwalpl) [reply py-utils/src/utils/message_bus/message_broker_collection.py:405]: Exception needs to be raised and handled by the caller as standard mechanism.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 2 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py:297]: That is the current behavior, It will fail if it reaches the retry_count.

Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py]: Done. Retried for 20 counts and it will fail it it reach beyond the count

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/src/setup/utils.py [hunk 3]
```diff
@@ -294,7 +296,19 @@ def reset():
...
+                                f" Unable to delete messages for message type" \
+                                f" {message_type} after {retry_count} retries")
+                            raise MessageBusError(errors.ERR_OP_FAILED,\
+                                "Unable to delete messages for message type" + \
+                                "%s after %d retries", message_type, \
+                                retry_count)
+                        rc = producer.delete()
+                        if rc == 0:
+                            break
+                        time.sleep(2*retry_count)
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:311]: Remove hard coded value 2 and replace with variable called wait_factor and set that value as 0. 
See if you can make retry_count value configurable from outside. 

def reset(retry_count):
....

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

**Review by ujjwalpl — APPROVED** [review]

**Comment by sachinpunadikar** [comment]
test this please

**Comment by sachinpunadikar** [comment]
run vm-deployment

**Comment by sachinpunadikar** [comment]
test this please

**Comment by sachinpunadikar** [comment]
run vm-deployment

**Comment by sachinpunadikar** [comment]
rerun vm-deployment

**Pull request merged**